plugin/file/metrics.go

```diff
@@ -0,0 +1,54 @@
+package file
+
+import (
+	"github.com/coredns/coredns/plugin"
+
+	"github.com/prometheus/client_golang/prometheus"
+)
+
+// Metrics for the secondary plugin
+var (
+	UpdatedSeconds = prometheus.NewHistogramVec(prometheus.HistogramOpts{
+		Namespace: plugin.Namespace,
+		Subsystem: "secondary",
+		Name:      "last_updated_seconds",
+		Buckets:   plugin.TimeBuckets,
+		Help:      "Histogram of the duration of an update",
+	}, []string{"zone"})
+
+	UpdatedSerial = prometheus.NewGaugeVec(prometheus.GaugeOpts{
+		Namespace: plugin.Namespace,
+		Subsystem: "secondary",
+		Name:      "last_updated_serial",
+		Help:      "Gauge of the last updated serial",
+	}, []string{"zone"})
+
+	UpdatedSuccess = prometheus.NewGaugeVec(prometheus.GaugeOpts{
+		Namespace: plugin.Namespace,
+		Subsystem: "secondary",
+		Name:      "last_updated_success",
+		Help:      "Gauge of the last updated timestamp",
+	}, []string{"zone"})
+
+	RefreshCount = prometheus.NewCounterVec(prometheus.CounterOpts{
+		Namespace: plugin.Namespace,
+		Subsystem: "secondary",
+		Name:      "refresh_total",
+		Help:      "Counter of the number of refreshes",
+	}, []string{"zone"})
+
+	RefreshSeconds = prometheus.NewHistogramVec(prometheus.HistogramOpts{
+		Namespace: plugin.Namespace,
+		Subsystem: "secondary",
+		Name:      "refresh_seconds_total",
```

Comment:
Histograms don't need `_total`.

```suggestion
		Name:      "refresh_seconds",
```